pilot/pkg/proxy/envoy/v2/debug.go

```diff
@@ -206,6 +209,68 @@ func (s *DiscoveryServer) endpointz(w http.ResponseWriter, req *http.Request) {
 	_, _ = fmt.Fprint(w, "\n{}]\n")
 }
 
+// SyncedVersions shows what resourceVersion of a given resource has been acked by Envoy.
+type SyncedVersions struct {
+	ProxyID         string `json:"proxy,omitempty"`
+	ClusterVersion  string `json:"cluster_acked,omitempty"`
+	ListenerVersion string `json:"listener_acked,omitempty"`
+	RouteVersion    string `json:"route_acked,omitempty"`
+}
+
+func (s *DiscoveryServer) distributedVersions(w http.ResponseWriter, req *http.Request) {
+	if !features.EnableDistributionTracking {
+		w.WriteHeader(http.StatusConflict)
+		_, _ = fmt.Fprint(w, "Pilot Version tracking is disables.  Please set the "+
+			"PILOT_ENABLE_DISTRIB_TRACKING environment varialbe to true to enable.")
+	}
+	if resourceID := req.URL.Query().Get("resource"); resourceID != "" {
```

Comment:
add a namespace filter. Perhaps even a nodeId filter or something

Reply:
I'm not clear on what is meant by nodeId.  The design doc lists one query string parameter, which includes the fully qualified name of the resource.  Are you suggesting we break this into namespace, type, and name parameters?

Reply:
Yes because imagine we have 1000 proxies. The user typically wants to know about the state of a resource for a specific set of proxies of a client-service. or for proxies in a namespace. Not for all proxies in the system. So it feels wasteful to lock down pilot data structures while you query the xds state in every proxy